### DISCUSSION TITLE
Filament 3 select validation error message hide issue

### DISCUSSION THREAD

**Discussion by midhunraaj** [filamentphp/filament#13989] [General]
I'm encountering an issue with the Laravel Filament 3 select field. When the field is set to "required," the error message appears as expected, but it does not disappear even after a value is selected. Is there a solution to this problem?

**Comment by dissto** [comment]
You probably need to make your select [reactive](https://filamentphp.com/docs/3.x/forms/advanced#the-basics-of-reactivity), since the next time the validation gets triggered is when you submit the form again 🤔

```php
Select::make('status')
  ->options(...)
  ->live()
  ```

**Comment by samuelhgf** [comment]
If you are using the `public ?array $data = [];` pattern, make sure to have the fields inside there as well like:
```php
public ?array $data = [
    'user_id' => null,
];
```

And for the live validation work, do this in any field
```php
->afterStateUpdated(function ($livewire, $component) {
    $livewire->validateOnly($component->getStatePath());
})
```

like...
```php
Select::make('user_id')
    ->label('User')
    ->searchable()
    ->options(User::all()->pluck('name', 'id'))
    ->required()
    ->live()
    ->afterStateUpdated(function ($livewire, $component) {
        $livewire->validateOnly($component->getStatePath());
    })
```
